hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -1376,6 +1441,8 @@ update env ledger st ev = case (st, ev) of
     onOpenClientDecommit headId ledger currentSlot coordinatedHeadState decommitTx
   (Open openState, NetworkInput ttl (ReceivedMessage{msg = ReqDec{transaction}})) ->
     onOpenNetworkReqDec env ledger ttl openState transaction
+  -- FIXME! check if we should observe deposits on close
+  -- FIXME! check what happens if we observe a deposit after fanout
```

Comment:
Answer: yes, we should track deposits in all states and allow recovering of them in all states.

The requirements in https://github.com/cardano-scaling/hydra/issues/1812 are incomplete and should mention that.